### DISCUSSION TITLE
recv: 550-Please turn on SMTP Authentication in your mail client

### DISCUSSION THREAD

**Discussion by cuongd86** [zytzagoo/smtp-validate-email#73] [Q&A]
Hi guys,
I'm using this lib to validate email on my site and I get this error:
`recv: 550-Please turn on SMTP Authentication in your mail client`

What should I do to fix it? You can check log file in the attachment.
[log.txt](https://github.com/zytzagoo/smtp-validate-email/files/6794913/log.txt)

**Comment by zytzagoo** [comment]
Hi, see https://github.com/zytzagoo/smtp-validate-email/issues/71 -- you need to implement https://en.wikipedia.org/wiki/SMTP_Authentication

**Reply by cuongd86** [reply]
thanks for your reply. Could you show me where I need to implement SMTP Authentication? I'm a newbie and I don't know how to do it.

**Reply by zytzagoo** [reply]
You'd need to add code to handle all the various parts of it:
- handle/parse server's announcement of supported auth protocols
- add methods to implement each of the possible auth protocols
- implement a way to provide/specify auth credentials per email/domain/server
- extend ehlo/helo smtp command flow to do/perform auth based on server's announced/preferred protocol (using the credentials given for that email/server)

It's not that hard, but it requires some work. 

Inspiration for auth protocol parts/details can be gleaned from Net_SMTP Pear class for example... Or PHPMailer maybe too.

**Reply by cuongd86** [reply]
Thank you @zytzagoo. I will try it

**Reply by cuongd86** [reply]
@zytzagoo SMTP authentication works and I validate some non existent mailboxes but I still get this response:

[2021-08-16T01:09:42.297184+0000] <<<recv:235 Authentication succeeded

[2021-08-16T01:09:42.297215+0000] send>>>: MAIL FROM:<1@******.net>

[2021-08-16T01:09:42.297569+0000] <<<recv: 250 OK

[2021-08-16T01:09:42.297609+0000] send>>>: NOOP

[2021-08-16T01:09:42.297843+0000] <<<recv: 250 OK

[2021-08-16T01:09:42.297868+0000] send>>>: NOOP

[2021-08-16T01:09:42.298085+0000] <<<recv: 250 OK

[2021-08-16T01:09:42.298117+0000] send>>>: RCPT TO:<peter@*******.com>

[2021-08-16T01:09:42.317116+0000] <<<recv: 250 Accepted

Could you tell me what do I miss?

**Reply by zytzagoo** [reply]
Not sure what you mean by "miss"? Every server is different... If you mean that you do not like that it returned "accepted" (for a non-existing address?), that's just the way it is. You can try changing the order of commands (to avoid being recognised via heuristics to be one of the "detectors"), or extend it to maybe try to start sending mail body (and hoping that maybe the server rejects the recipient at that point)... But nothing really prevents the server to just appear to accept email for non-existing recipients (and then just null/void the data you just sent it after you disconnect). There is no way to be 100% certain about an email addreess existing (or not) "from the outside".

Or was the question about something else?
